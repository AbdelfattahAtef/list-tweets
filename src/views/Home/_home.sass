@import "../../styles/_variables.sass"
@import "../../styles/_breakpoints.sass"

.home-wrapper
  &__content
    display: flex
    justify-content: space-between
    +breakpoint(xsmax)
      margin-top: 50px
