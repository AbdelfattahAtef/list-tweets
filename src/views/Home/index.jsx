import React, { Component } from 'react';
import './_home.sass';

import Header from "../../components/Header";
import Navbar from "../../components/Navbar";
import Sidebar from "../../components/Sidebar";
import Tweets from "../../components/Tweets";

class Home extends Component {
    state = {
        toggleSidebar: false,
        tweetsNumber: 10,
    };

    /**
     * Toggle display of sidebar
     * @param displaySidebar
     */
    toggleSidebar = (displaySidebar) => {
        this.setState({
            toggleSidebar: displaySidebar
        })
    };

    numberOfTweets = (value) => {
        debugger
        this.setState({
            tweetsNumber: value && value <= 10 ? value : 10,
        })
    };

    render() {
        return (
            <div className="home-wrapper">
                <Navbar toggleSidebar={this.toggleSidebar}/>
                <Header/>
                <div className="home-wrapper__content">
                    <Sidebar
                        toggleSidebar={this.state.toggleSidebar}
                        numberOfTweets={this.numberOfTweets}/>
                    <Tweets tweetsNumber={this.state.tweetsNumber}/>
                </div>
            </div>
        );
    }
}

export default Home;
