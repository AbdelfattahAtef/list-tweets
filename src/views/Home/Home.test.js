import React from 'react';
import Home from '../Home';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';

configure({ adapter: new Adapter() });

describe('<Home />', () => {
    it('should render Home component', () => {
        const wrapper = shallow(<Home />);
        expect(wrapper.find('.home-wrapper').length).toBe(1);
        expect(wrapper.find('.home-wrapper__content').length).toBe(1);
    });

    it('calls componentDidMount', () => {
        jest.spyOn(Home.prototype, 'componentDidMount');
        const wrapper = shallow(<Home />);
        expect(Home.prototype.componentDidMount.mock.calls.length).toBe(1)
    });

    it('matches the snapshot', () => {
        const tree = shallow(<Home />);
        expect(toJson(tree)).toMatchSnapshot();
    });
});
