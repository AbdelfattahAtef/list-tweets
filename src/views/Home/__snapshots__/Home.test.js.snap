// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Home /> matches the snapshot 1`] = `
<Fragment>
  <div
    className="home-wrapper"
  >
    <Navbar
      toggleSidebar={[Function]}
    />
    <Header />
    <div
      className="home-wrapper__content"
    >
      <Sidebar
        numberOfTweets={[Function]}
        toggleSidebar={false}
      />
      <Tweets
        tweetsNumber={10}
      />
    </div>
  </div>
  <Loader />
</Fragment>
`;
